### PR TITLE
initial btcd commit

### PR DESCRIPTION
**Pull request by RandyMcMillan**

### CODE REVIEW

**Comment by RandyMcMillan** [comment]
Screen Shot 2021-09-12 at 6 06 08 PM

**Comment by RandyMcMillan** [comment]
This btcd PR is anticipation for offering optional features to plebnet playground users.

**Comment by rsafier** [comment]
could you refactor this to have the docker container actually just pull the code inside itself and build, don't want to have to carry around all their source in the repo, should be in a docker build stage

**Comment by RandyMcMillan** [comment]
good to go...

**Comment by rsafier** [comment]
> good to go...

could you refactor this to have the docker container actually just pull the code inside itself and build, don't want to have to carry around all their source in the repo, should be in a docker build stage

**Comment by RandyMcMillan** [comment]
good point - will do

File: GNUmakefile [hunk 1]
```diff
@@ -267,27 +267,6 @@ ifneq ($(shell id -u),0)
 #.ONESHELL:
 	sudo -s
 endif
```
Comment (RandyMcMillan) [inline GNUmakefile:269]: Removed a lot of junk

File: GNUmakefile [hunk 2]
```diff
@@ -315,88 +294,25 @@ endif
...
-	$(DOCKER_COMPOSE) $(VERBOSE) -p $(PROJECT_NAME)_$(HOST_UID) run --rm shell sh
-	@echo ''
-else
-	# run the command
-	@echo ''
-	$(DOCKER_COMPOSE) $(VERBOSE) -p $(PROJECT_NAME)_$(HOST_UID) run --rm shell sh -c "$(CMD_ARGUMENTS)"
-	@echo ''
-endif
-
 .PHONY: run
```
Comment (RandyMcMillan) [inline GNUmakefile:304]: Removed junk

File: GNUmakefile [hunk 3]
```diff
@@ -315,88 +294,25 @@ endif
...
-ifneq ($(shell id -u),0)
-	bash -c 'sudo make torproxy user=root &'
-endif
-ifeq ($(CMD_ARGUMENTS),)
-	$(DOCKER_COMPOSE) $(VERBOSE) -f docker-compose.yml -p $(PROJECT_NAME)_$(HOST_UID) run --publish 8118:8118 --publish 9050:9050  --publish 9051:9051 --rm torproxy
-else
-	$(DOCKER_COMPOSE) $(VERBOSE) -f docker-compose.yml -p $(PROJECT_NAME)_$(HOST_UID) run --publish 8118:8118 --publish 9050:9050  --publish 9051:9051 --rm torproxy sh -c "$(CMD_ARGUMENTS)"
-endif
-	@echo ''
 #######################
```
Comment (RandyMcMillan) [inline GNUmakefile:316]: Removed junk

File: GNUmakefile [hunk 4]
```diff
@@ -407,32 +323,14 @@ clean:
...
-	bash -c "cat ./docker/README.md >  README.md"
-	bash -c "cat ./docker/DOCKER.md >> README.md"
-#	bash -c "echo '<insert string>' >> README.md"
-	bash -c "echo '----' >> README.md"
-	bash -c "echo '## [$(PROJECT_NAME)]($(GIT_SERVER)/$(GIT_PROFILE)/$(PROJECT_NAME)) [$(GIT_HASH)]($(GIT_SERVER)/$(GIT_PROFILE)/$(PROJECT_NAME)/commit/$(GIT_HASH))' >> README.md"
-	bash -c "echo '##### &#36; <code>make</code>' >> README.md"
-	bash -c "make report >> README.md"
-	bash -c "make help   >> README.md"
-	bash -c "if hash open 2>/dev/null; then open README.md; fi || echo failed to open README.md"
 .PHONY: push
```
Comment (RandyMcMillan) [inline GNUmakefile:334]: Removed junk

File: btcd/btcd.conf [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+signet=1
+rpcuser=bitcoin
+rpcpass=c8c8b9740a470454255b7a38d4f38a52$e8530d1c739a3bb0ec6e9513290def11651afbfd2b979f38c16ec2cf76cf348a
+txindex=1
+nodnsseed=1
+#generate=1
+rpclisten=0.0.0.0:38332
+whitelist=0.0.0.0/0
+signetchallenge=512102ee856c56a5aaadd1656f849bafa4c9dacc86a2878fe546c6189185f842ae2c1851ae
+addpeer=104.131.10.218:38333
```
Comment (RandyMcMillan) [inline btcd/btcd.conf:11]: We stay with similar settings as bitcoin-signet

File: GNUmakefile [hunk 6]
```diff
@@ -464,9 +362,5 @@ ifeq ($(slim),true)
 	make package-all slim=false
 endif
 	make header package-header build package-statoshi
-
-
 ########################
--include funcs.mk
--include Makefile
```
Comment (RandyMcMillan) [inline GNUmakefile:366]: Removed junk

**Comment by RandyMcMillan** [comment]
rebased on top of #24

**Pull request merged**